### PR TITLE
Medical - Add API for fully healing a unit

### PR DESCRIPTION
**Pull request by DartRuffian**
**When merged this pull request will:**
- Add a public API function to full heal a unit
  - This is just a wrapper with some param checking
- Add optional parameter to `medical_treatment_fnc_fullHeal` to not add the full heal log message

### IMPORTANT

- If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: addons/medical/functions/fnc_fullHeal.sqf [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ *
+ * Example:
+ * [player, cursorObject] call ace_medical_fnc_fullHeal
+ *
+ * Public: Yes
+ */
+
+params [["_medic", objNull, [objNull]], ["_patient", objNull, [objNull]]];
+
+if (isNull _medic or !alive _patient) exitWith {
```
Comment (DartRuffian) [inline addons/medical/functions/fnc_fullHeal.sqf:21]: Should dead medics be allowed to heal?

I couldn't tell if a dead medic would cause any issues from a quick look, but I'm not experienced with medical's internals

Comment (DartRuffian) [reply addons/medical/functions/fnc_fullHeal.sqf:21]: Or alternatively, maybe the parameters should be swapped and _medic be made optional, so you could just do `[_unit] call ace_medical_fnc_fullHeal` in cases where there is no medic?

File: addons/medical/functions/fnc_fullHeal.sqf [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    ERROR_2("fullHeal [medic %1][patient %2] Bad parameters",_medic,_patient);
+};
+
+[_medic, _patient] call EFUNC(medical_treatment,fullHeal);
```
Comment (DartRuffian) [inline addons/medical/functions/fnc_fullHeal.sqf:25]: Thought about adding a boolean flag to fullHeal to enable/disable the full heal message that shows in the medical GUI. Should that be a separate PR after this or just do it in one?

File: addons/medical/functions/fnc_fullHeal.sqf [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ * [player, cursorObject] call ace_medical_fnc_fullHeal
+ *
+ * Public: Yes
+ */
+
+params [["_medic", objNull, [objNull]], ["_patient", objNull, [objNull]]];
+
+if (isNull _medic or !alive _patient) exitWith {
+    ERROR_2("fullHeal [medic %1][patient %2] Bad parameters",_medic,_patient);
+};
```
[the remark this answers is not in the document]
Comment (DartRuffian) [reply addons/medical/functions/fnc_fullHeal.sqf:23]: Yeah that's what I was thinking of changing it to, left a comment about it but I'll just use your suggestion

**Comment by DartRuffian** [comment]
> Thought about adding a boolean flag to `medical_treatment_fnc_fullHeal` to enable/disable the full heal message that shows in the medical GUI. Should that be a separate PR after this or just do it in one?

Figured I'd just include an example of what I meant, commit can just be reverted if it should be a separate PR

**Review by PabstMirror — APPROVED** [review]

File: addons/medical/functions/fnc_fullHeal.sqf [hunk 4]
```diff
@@ -0,0 +1,31 @@
+#include "..\script_component.hpp"
+/*
+ * Author: DartRuffian
+ * Fully heals the patient.
+ *
+ * Arguments:
+ * 0: Patient <OBJECT>
+ * 1: Medic <OBJECT><OPTIONAL>
+ * 2: Log message <BOOL><OPTIONAL>
```
[the remark this answers is not in the document]
Comment (DartRuffian) [reply addons/medical/functions/fnc_fullHeal.sqf:9]: That's how I thought ace did it, but looking at other functions just showed them using `<OPTIONAL>`

File: addons/medical/functions/fnc_fullHeal.sqf [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+if (isNull _medic) then {
+    _medic = _patient;
+};
+
+if (!alive _patient) exitWith {
+    ERROR_2("fullHeal [medic %1][patient %2] Bad parameters",_medic,_patient);
+};
+
+[_medic, _patient, _logMessage] call EFUNC(medical_treatment,fullHeal);
+nil
```
Comment (johnb432) [inline addons/medical/functions/fnc_fullHeal.sqf:31]: Function returns nothing, so useless to explicitly return nothing.
```suggestion
```

File: addons/medical/functions/fnc_fullHeal.sqf [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+if (isNull _medic) then {
+    _medic = _patient;
+};
+
+if (!alive _patient) exitWith {
+    ERROR_2("fullHeal [medic %1][patient %2] Bad parameters",_medic,_patient);
+};
+
+[_medic, _patient, _logMessage] call EFUNC(medical_treatment,fullHeal)
```
Comment (johnb432) [inline addons/medical/functions/fnc_fullHeal.sqf:30]: When it returns `nil`, we leave the semicolon.
```suggestion
[_medic, _patient, _logMessage] call EFUNC(medical_treatment,fullHeal);
```

**Review by johnb432 — APPROVED** [review]

**Review by BaerMitUmlaut — CHANGES_REQUESTED** [review]

File: addons/medical/functions/fnc_fullHeal.sqf [hunk 7]
```diff
@@ -0,0 +1,30 @@
+#include "..\script_component.hpp"
+/*
+ * Author: DartRuffian
+ * Fully heals the patient.
+ *
+ * Arguments:
+ * 0: Patient <OBJECT>
+ * 1: Medic <OBJECT> (default: objNull)
+ * 2: Log message <BOOL> (default: true)
```
[the remark this answers is not in the document]
Comment (DartRuffian) [reply addons/medical/functions/fnc_fullHeal.sqf:9]: I only left it default true since that's what the default behavior was. The public version could default to false though

File: addons/medical_treatment/functions/fnc_fullHeal.sqf [hunk 8]
```diff
@@ -16,8 +17,10 @@
  * Public: No
  */
 
-params ["_medic", "_patient"];
+params ["_medic", "_patient", ["_showMessage", true]];
```
[the remark this answers is not in the document]
Comment (DartRuffian) [reply addons/medical_treatment/functions/fnc_fullHeal.sqf:20]: I had it as `_logMessage` in the API function, I guess I named them different on accident

File: addons/medical/functions/fnc_fullHeal.sqf (hunk 7 above)
[the remark this answers is not in the document]
Comment (DartRuffian) [reply addons/medical/functions/fnc_fullHeal.sqf:9]: > This also doesn't note where the log ends up.
Should it be like "Log message to medical logs"?

[the remark this answers is not in the document]
Comment (BaerMitUmlaut) [reply addons/medical/functions/fnc_fullHeal.sqf:9]: Maybe "Write message to patient log"
